include/eigenpy/std-array.hpp

```diff
@@ -0,0 +1,140 @@
+/// Copyright (c) 2023 CNRS INRIA
+
+#ifndef __eigenpy_utils_std_array_hpp__
+#define __eigenpy_utils_std_array_hpp__
+
+#include <boost/python/suite/indexing/indexing_suite.hpp>
+#include "eigenpy/std-vector.hpp"
+
+#include <array>
+
+namespace eigenpy {
+
+template <typename Container, bool NoProxy, class SliceAllocator,
+          class DerivedPolicies>
+class array_indexing_suite;
+namespace details {
+
+template <typename Container, bool NoProxy, class SliceAllocator>
+class final_array_derived_policies
+    : public array_indexing_suite<
+          Container, NoProxy, SliceAllocator,
+          final_array_derived_policies<Container, NoProxy, SliceAllocator> > {};
+}  // namespace details
+
+template <typename Container, bool NoProxy = false,
+          class SliceAllocator = std::allocator<typename Container::value_type>,
+          class DerivedPolicies = details::final_array_derived_policies<
+              Container, NoProxy, SliceAllocator> >
+class array_indexing_suite
+    : public bp::vector_indexing_suite<Container, NoProxy, DerivedPolicies> {
+ public:
+  typedef typename Container::value_type data_type;
+  typedef typename Container::value_type key_type;
+  typedef typename Container::size_type index_type;
+  typedef typename Container::size_type size_type;
+  typedef typename Container::difference_type difference_type;
+  typedef std::vector<data_type, SliceAllocator> slice_vector_type;
+  static constexpr std::size_t Size = std::tuple_size<Container>{};
+
+  template <class Class>
+  static void extension_def(Class &) {}
+
+  template <class Iter>
+  static void extend(Container &, Iter, Iter) {}
+
+  static void append(Container &, data_type const &) {}
+  // no-op
+  static void delete_item(Container &, index_type) {}
+  // no-op
+  // no-op
+  static void delete_slice(Container &, index_type, index_type) {}
+
+  static void set_slice(Container &container, index_type from, index_type to,
+                        data_type const &v) {
+    if (from > to) {
+      return;
+    } else {
+      std::fill(container.begin() + from, container.begin() + to, v);
+    }
+  }
+
+  template <class Iter>
+  static void set_slice(Container &container, index_type from, index_type to,
+                        Iter first, Iter last) {
+    if (from > to) {
+      return;
+    } else {
+      std::copy(first, last, container.begin() + from);
+    }
+  }
+
+  static bp::object get_slice(Container &container, index_type from,
+                              index_type to) {
+    if (from > to) return bp::object(std::array<data_type, 0>());
```

Comment:
This function has two case :
- In one case, `get_slice` return a std::array
- In the nominal case, `get_slice` return a std::vector

I think this behavior could be misleading. In some case, it will be possible to modify the slice length, and in other case, it will not be possible.

Reply:
You're right. Ideally we would return some type of container that has dynamic (runtime-known) size but isn't resizable.

Reply:
Thinking about it further, I think the best option would be to return a boost::span object (since we are not C++20 yet and don't have std::span). Which means exposing boost::span.